engines/router/missionctl/fiberapi/traffic_splitting_strategy.go

```diff
@@ -109,47 +110,33 @@ func (s *TrafficSplittingStrategy) SelectRoute(
 	ctx context.Context,
 	req fiber.Request,
 	routes map[string]fiber.Component,
-) (fiber.Component, []fiber.Component, error) {
-	var orderedRoutes []fiber.Component
-	// array, that holds results of testing the request by each rule configured on the strategy
-	// `results[k]` – is `true` if the request satisfies `k`th rule of the strategy, and `false`
-	// otherwise
-	results := make([]bool, len(s.Rules))
-
-	for idx, rule := range s.Rules {
+) (fiber.Component, []fiber.Component, fiber.Labels, error) {
+	labels := fiber.NewLabelsMap()
+
+	for _, rule := range s.Rules {
 		res, err := rule.TestRequest(req)
 		if err != nil {
-			return nil, nil, createFiberError(err, req.Protocol())
-		}
-
-		results[idx] = res
-	}
-
-	// select primary route and fallbacks, based on the results of testing
-	// given request against traffic-splitting rules
-	for i := 0; i < len(results); i++ {
```

Comment:
I might have some misconception here, 

I would had thought because of the orthogonality checks on the rules, there should always be at most one rule that can be matched given a request, and that should be the reason of the removal (and probably previously), here instead of 

> we may not have enough time to (sequentially) execute multiple matches

didnt catch the timing part since to me there should only be 1 rule matched for any request.

Reply:
Ahh you're right about your assumption but it's missing one detail - the idea of overlapping rules at different priorities (kind of similar to XP experiments). [This](https://github.com/caraml-dev/turing/blob/main/api/turing/validation/validator.go#L262) is the relevant validation.

So, if there were 2 rules:
* {"a": [1]}
* {"a": [1], "b": [2]}
A request that matches # 2, will also match # 1. In this case, we will just go with the order in which the rules are defined, to pick the appropriate rule (per this example, # 2 will never be applied because it comes after the broader rule).